lib/tty/pager.rb

```diff
@@ -1,156 +1,79 @@
 # frozen_string_literal: true
 
-require 'tty-screen'
-
-require_relative 'pager/basic'
-require_relative 'pager/null'
-require_relative 'pager/system'
-require_relative 'pager/version'
+require_relative "pager/basic"
+require_relative "pager/null"
+require_relative "pager/system"
+require_relative "pager/version"
 
 module TTY
-  class Pager
+  module Pager
     Error = Class.new(StandardError)
     PagerClosed = Class.new(Error)
 
-    # Select an appriopriate pager
-    #
-    # If the user disabled paging then a NullPager is returned,
-    # otherwise a check is performed to find native system
-    # command to perform pagination with SystemPager. Finally,
-    # if no system command is found, a BasicPager is used which
-    # is a pure Ruby implementation known to work on any platform.
-    #
-    # @api private
-    def self.select_pager(enabled, commands)
-      if !enabled
-        NullPager
-      elsif SystemPager.exec_available?(*commands)
-        SystemPager
-      else
-        BasicPager
+    module ClassMethods
+      # Create a pager
+      #
+      # @param [Hash] options
+      # @option options [Proc] :prompt
+      #   a proc object that accepts page number
+      # @option options [IO] :input
+      #   the object to send input to
+      # @option options [IO] :output
+      #   the object to send output to
+      # @option options [Boolean] :enabled
+      #   disable/enable text paging
+      #
+      # @api public
+      def new(**options)
+        pager_klass = select_pager(**options)
+        @pager = pager_klass.new(**options)
```

Comment:
Shouldn't this directly return the results of `pager_klass.new` instead of saving it in an instance variable? How is that ivar used later?

Reply:
Nicely spotted - remnants from the gradual refactor.